bokeh/models/plots.py

```diff
@@ -164,7 +164,7 @@ def add_glyph(self, source_or_glyph, glyph=None, **kw):
     y_mapper_type = Either(Auto, String)
 
     extra_x_ranges = Dict(String, Instance(Range1d))
-    extra_y_ranges = Dict(String, Instance(Range1d))
+    extra_y_ranges = Dict(String, Instance(Range))
```

Comment:
I can't get why you need to change plot definition directly? I'm not sure but I think you should be be doing this on the charts level and not changing plots code..